docs/howto/advanced/cancellation.md

```diff
@@ -54,22 +54,8 @@ def my_task(context):
     do_something_expensive()
 ```
 
-There is also an async API
-
-```python
-@app.task(pass_context=True)
-async def my_task(context):
-  for i in range(100):
-    if await context.should_abort_async():
-      raise exceptions.JobAborted
-    do_something_expensive()
-```
-
-:::{warning}
-`context.should_abort()` and `context.should_abort_async()` does poll the
-database and might flood the database. Ensure you do it only sometimes and
-not from too many parallel tasks.
-:::
+The worker receives a Postgres notification every time a running job is requested to abort, unless `listen_notify=False`.
```

Comment:
How about adding something like "Internally, the worker receives ..."? It would make it clearer that this is nothing the user has to handle but something that works behind the curtain.

We should also update the docstrings of [`listen_notify`](https://github.com/onlyann/procrastinate/blob/9e4e0dcb140fcdc56be3817e4a963aa222fde77a/procrastinate/app.py#L271) and [`polling_interval`](https://github.com/onlyann/procrastinate/blob/9e4e0dcb140fcdc56be3817e4a963aa222fde77a/procrastinate/app.py#L261-L262) that this is not just for requesting jobs.

So, `should_abort()` is now used for async and sync tasks. Correct? Haven't you originally planned to cancel the async tasks automatically on abort? (The thing I was a bit ambivalent about). Just wondering.

Reply:
> So, should_abort() is now used for async and sync tasks. Correct? 

Correct

> Haven't you originally planned to cancel the async tasks automatically on abort?

Yes. I aim to keep this PR focused or notification support and open another about issuing an asyncio cancel on async tasks.

Reply:
I amended the documentation.
I also noticed I need to rewrite some of the discussion section around the worker. I will however leave that part for another PR as it is not related to abort notification.

Reply:
Sure, but maybe we can have updated docstrings for [`listen_notify`](https://github.com/onlyann/procrastinate/blob/9e4e0dcb140fcdc56be3817e4a963aa222fde77a/procrastinate/app.py#L271) and [`polling_interval`](https://github.com/onlyann/procrastinate/blob/9e4e0dcb140fcdc56be3817e4a963aa222fde77a/procrastinate/app.py#L261-L262).

Currently, it says for `polling_interval`:
```markdown
polling_interval : ``float``
    Indicates the maximum duration (in seconds) the worker waits between
    each database job poll. Raising this parameter can lower the rate at which
    the worker makes queries to the database for requesting jobs.
    (defaults to 5.0)
```
How about:
```markdown
polling_interval : ``float``
    Indicates the maximum duration (in seconds) the worker waits between
    each database job poll. Raising this parameter can lower the rate at which
    the worker makes queries to the database for requesting jobs or checking
    for job abortion requests.
    (defaults to 5.0)
```

And for `listen_notify` it says:
```markdown
listen_notify : ``bool``
    If ``True``, the worker will dedicate a connection from the pool to
    listening to database events, notifying of newly available jobs.
    If ``False``, the worker will just poll the database periodically
    (see ``polling_interval``). (defaults to ``True``)
```
How about:
```markdown
listen_notify : ``bool``
    If ``True``, the worker will dedicate a connection from the pool to
    listening to database events, notifying of newly available jobs, or
    job abortion requests.
    If ``False``, the worker will just poll the database periodically
    (see ``polling_interval``). (defaults to ``True``)
```

Reply:
(while we're at it, 
```
If ``False``, the worker will just poll the database periodically
```
might be misleading in implying that this doesn't happen with ``True``
Maybe:
> If `False`, the worker won't listen to database events, so it will only
> know about new jobs & abortion requests through the polling mechanism
```
?

Reply:
Pushing some update on this from your feedbacks.
Let me know what you think

Reply:
What does the bullet point `Fetching updates for running jobs` mean? Does the polling more than checking for new jobs and abortion requests?